src/client/main.cpp

```diff
@@ -41,7 +42,12 @@ int main(int argc, char** argv) {
             strategy = static_cast<BotStrategy>(result["strategy"].as<int>());
         }
 
-        ClientGameController client_game_controller(is_headless, is_bot, strategy, server_ip);
+        bool tcp_port_opt_present = result.count("port");
+        uint32_t tcp_port = GAME_SETTINGS.SERVER_TCP_PORT;
+        if (tcp_port_opt_present) {
```

Comment:
I would just replace this with `result.count("port")` instead of using the extra var `tcp_port_opt_present`